internal/mail/mail.go

```diff
@@ -78,6 +78,10 @@ func SendIndividualMail(to []server.EmailRecipient, from server.EmailSender, cc
 		}
 	}
 
+	if len(tags) > 0 {
+		sendMessage.AddCategories(tags...)
```

Comment:
i wonder if they have a key value pair instead of just list of strings

Reply:
![image](https://github.com/commitdev/zero-notification-service/assets/1787590/bd402897-9c3f-4627-a7f3-253337dc3f92)
like idonno if this is per request

Reply:
they also have `custom_args`
![image](https://github.com/commitdev/zero-notification-service/assets/1787590/d0138909-1c98-4826-9fbf-d48d68bf8477)
i wonder if its a better place to store this

Reply:
looks like theres some limitations https://docs.sendgrid.com/for-developers/sending-email/categories
![image](https://github.com/commitdev/zero-notification-service/assets/1787590/6ced2932-3e35-420e-bcc5-caded3ceeef7)

Reply:
hmm `custom_args` would allow us to add new params without affecting the function params

Reply:
yeah if your tag is the string `recommendation-email`  then its good, is that what we want?

Reply:
if the string is a `<uuid>` then custom args is what we want i think

Reply:
we don't need to segment by user id or anything on sendgrid so i think categories is fine, for uuid level events, prob use amplitude

I don't think we need to tie the amplitude behavior back to sendgrid open rates, and I think we can track the CTA event on the actual email button click->page load

we do need to track what jobs got sent out but I think we only need an email id from sendgrid or something (not sure if this exists)

probably the most we need is multiple tags for now

Reply:
okay, i thought we wanted to track userId <-> email opens

Reply:
if we just want to group the emails open rate this is good

Reply:
actually wouldn't hurt to put it in since this is a lib

Reply:
nice thanks